### PR TITLE
Expose dilated convolutions

### LINKED ISSUE #1225: Feature Request: Add dilate to ConvTranspose functions
Any possibility of adding dilatation to transpose convolutions?

### PR DESCRIPTION
**Pull request by ezyang**
Fixes #1225.

There's also a documentation fixup patch in the series.

CC @soumith

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Comment by soumith** [comment]
thanks Ed!

File: torch/nn/modules/conv.py [hunk 1]
```diff
@@ -506,11 +507,11 @@ class ConvTranspose2d(_ConvTransposeMixin, _ConvNd):
     """
 
     def __init__(self, in_channels, out_channels, kernel_size, stride=1,
-                 padding=0, output_padding=0, groups=1, bias=True):
+                 padding=0, output_padding=0, groups=1, bias=True, dilation=1):
         kernel_size = _pair(kernel_size)
         stride = _pair(stride)
         padding = _pair(padding)
-        dilation = _pair(1)
+        dilation = _pair(dilation)
```
Comment (hojonathanho) [inline torch/nn/modules/conv.py]: It seems that this dilation is never being passed into `F.conv_transpose2d` in `forward` below.

Comment (soumith) [inline torch/nn/modules/conv.py]: hmm you're right. cc: @ezyang to fix this.

Comment (hojonathanho) [inline torch/nn/modules/conv.py]: (the same is true for 1d and 3d too)

Comment (ezyang) [inline torch/nn/modules/conv.py]: Blorp, that's what I get for trusting the code without reading. I'll fix it tomorrow.

Comment (ezyang) [inline torch/nn/modules/conv.py]: See #1386, @hojonathanho

**Pull request merged**